src/gui/systray.cpp

```diff
@@ -133,123 +126,225 @@ void Systray::setToolTip(const QString &tip)
     QSystemTrayIcon::setToolTip(tr("%1: %2").arg(Theme::instance()->appNameGUI(), tip));
 }
 
-int Systray::calcTrayWindowX()
+bool Systray::syncIsPaused()
 {
-#ifdef Q_OS_OSX
-    // macOS handles DPI awareness differently
-    // and menu bar is always at the top, icons starting from the right
+    return _syncIsPaused;
+}
 
-    QPoint topLeft = this->geometry().topLeft();
-    QPoint topRight = this->geometry().topRight();
-    int trayIconTopCenterX = (topRight - ((topRight - topLeft) * 0.5)).x();
-    return trayIconTopCenterX - (400 * 0.5);
-#else
-    QScreen* trayScreen = nullptr;
-    if (QGuiApplication::screens().count() > 1) {
-        trayScreen = QGuiApplication::screens().at(screenIndex());
+void Systray::pauseResumeSync()
+{
+    if (_syncIsPaused) {
+        _syncIsPaused = false;
+        emit resumeSync();
     } else {
-        trayScreen = QGuiApplication::primaryScreen();
+        _syncIsPaused = true;
+        emit pauseSync();
     }
+}
 
-    int screenWidth = trayScreen->geometry().width();
-    int screenHeight = trayScreen->geometry().height();
-    int availableWidth = trayScreen->availableGeometry().width();
-    int availableHeight = trayScreen->availableGeometry().height();
-
-    QPoint topRightDpiAware = QPoint();
-    QPoint topLeftDpiAware = QPoint();
-    if (this->geometry().left() == 0 || this->geometry().top() == 0) {
-        // tray geometry is invalid - QT bug on some linux desktop environments
-        // Use mouse position instead. Cringy, but should work for now
-        topRightDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-    } else {
-        topRightDpiAware = this->geometry().topRight() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = this->geometry().topLeft() / trayScreen->devicePixelRatio();
+/********************************************************************************************/
+/* Helper functions for cross-platform tray icon position and taskbar orientation detection */
+/********************************************************************************************/
+
+QScreen *Systray::currentScreen() const
+{
+    const auto screens = QGuiApplication::screens();
+    const auto cursorPos = QCursor::pos();
+    const auto it = std::find_if(screens.cbegin(), screens.cend(), [=](QScreen *screen) {
+        return screen->geometry().contains(cursorPos);
+    });
+    return (it != screens.cend()) ? *it : nullptr;
```

Comment:
The same code of lines 153–156 with good old-fashioned range-based for-loop:
```c++
for (const auto screen : screens) {
    if (screen->geometry().contains(cursorPos)) return screen;
}
return nullptr;
```
Shorter (by number of characters), less complex, thus better readable.
I always wonder why people prefer trivial STL algorithms over even more trivial loops.

Reply:
I disagree on the reasoning. Shorter doesn't imply readable and more control flow means higher complexity for the function.

That being said, without ranges I agree the STL algorithms incur some unwanted noise by strong leakage of the iterator concept and that indeed has an impact on readability. I wish we had ranges with C++17 really...

As to why I prefer STL algorithms: in short, I've seen enough loops gone wrong in my career. Over time "simple loops" tend to have their body grow in complexity they are also tempting for "copy/paste/adjust" in both case you loose the intent quickly while algorithms force to keep that intent spelled out.

It's admittedly a bit better since we got for range loops. But that's merely a function of making them less verbose and less reliant on explicit iterator uses.

Still, I will change it and go for a for range loop. The function is focused enough that we can hope it won't degenerate as mentioned above.

Reply:
Ah, btw my comment was not a "please change this", just a "I always wonder why people prefer this". :-)

> Shorter doesn't imply readable

That's true in general, but in my specific case I think the range-based for loop is much more readable. Of course, readability is a subjective matter, but I would wonder if one seriously disagrees here.

> ranges

Ranges would delete iterator-based noise, but not `std::find_if` and lambda-based noise. What I want to say: even with ranges, the range-based for-loop is better regarding readability.

> I wish we had ranges with C++17 really...

Agreed, but, wouldn't even help here because the desktop client is compiled in C++14 mode, right?

> As to why I prefer STL algorithms: [...] algorithms force to keep that intent spelled out

That's indeed a good point.

> we can hope it won't degenerate as mentioned above

We have a review process to prevent that. :-)